tests/datasource/fluent/test_pandas_datasource.py

```diff
@@ -471,3 +471,24 @@ def test_dataframe_asset(empty_data_context: AbstractDataContext, test_df_pandas
             for asset in empty_data_context.sources.pandas_default.assets.values()
         ]
     )
+
+
+def test_dynamic_pandas_batch_metadata(
```

Comment:
More detail to be added to this test in my subsequent PR enabling config variables.